### PR TITLE
Enable OIDC as authentication

### LINKED ISSUE #153: [Enhancement][Nightly Playground] Enable OIDC for nightly playground
**Is your feature request related to a problem? Please describe.**
As of now, nightly playground has limited users based on local database (admin, kibanaserver, readonly, etc)
With this set up, it is difficult to manage the permissions privileges and audit the cluster for malicious activity. We want to be able to authenticate users and accordingly allow users to have advanced permissions to test out the cluster. 

**Describe the solution you'd like**
Enable Open ID Connect for nightly playground. The best way would be GitHub but since GitHub is not an identity provider, there might be other workarounds.

Option 1: Use third party IdP like [keycloak](https://www.keycloak.org/), [dex](https://dexidp.io/), etc to connect to GitHub
Option 2: User other method of authentication such as Google account (example: https://github.com/opensearch-project/dashboards-anywhere/issues/52), microsoft, etc.

**Describe alternatives you've considered**
Keep using local users 

**Additional context**
Try to implement a generic solution that can onboard any other project under opensearch-project with similar requirements.

cc: @prudhvigodithi @bbarani

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Integrated nightly playgrounds with keycloak which uses GH as an authentication mechanism. 

### Issues Resolved
closes #153 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: nightly-playground/resources/security-config/config.yml [hunk 1]
```diff
@@ -78,173 +78,56 @@ config:
...
+                93FcXmn/6pUCyziKrlA4b9v7LWIbxcceVOF34GfID5yHI9Y/QCB/IIDEgEw+OyQm
+                jgSubJrIqg0CAwEAAaNCMEAwDwYDVR0TAQH/BAUwAwEB/zAOBgNVHQ8BAf8EBAMC
+                AYYwHQYDVR0OBBYEFIQYzIU07LwMlJQuCFmcx7IQTgoIMA0GCSqGSIb3DQEBCwUA
+                A4IBAQCY8jdaQZChGsV2USggNiMOruYou6r4lK5IpDB/G/wkjUu0yKGX9rbxenDI
+                U5PMCCjjmCXPI6T53iHTfIUJrU6adTrCC2qJeHZERxhlbI1Bjjt/msv0tadQ1wUs
+                N+gDS63pYaACbvXy8MWy7Vu33PqUXHeeE6V/Uq2V8viTO96LXFvKWlJbYK8U90vv
+                o/ufQJVtMVT8QtPHRh8jrdkPSHCa2XV4cdFyQzR1bldZwgJcJmApzyMZFo6IQ6XU
+                5MsI+yMRQ+hDKXJioaldXgjUkK642M4UwtBV8ob2xJNDd2ZhwLnoQdeXeGADbkpy
+                rqXRfboQnoZsG4q5WTP468SQvvG5
+                -----END CERTIFICATE-----
```
Comment (peterzhuamazon) [inline nightly-playground/resources/security-config/config.yml:128]: This is a public key here right?

File: nightly-playground/resources/security-config/config.yml [hunk 2]
```diff
@@ -78,173 +78,56 @@ config:
...
+            subject_key: preferred_username
+            roles_key: roles
+            openid_connect_url: https://keycloak.opensearch.org/realms/opensearch-nightly-playgrounds/.well-known/openid-configuration
```
Comment (peterzhuamazon) [inline nightly-playground/resources/security-config/config.yml:131]: Are we ok to display the full url here?

File: nightly-playground/resources/security-config/config.yml (hunk 1 above)
Comment (gaiksaya) [reply nightly-playground/resources/security-config/config.yml:128]: Yes

File: nightly-playground/resources/security-config/config.yml (hunk 2 above)
Comment (gaiksaya) [reply nightly-playground/resources/security-config/config.yml:131]: Yes. Its a public domain

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**